app/angular/src/client/preview/types.ts

```diff
@@ -28,5 +28,5 @@ export interface StoryFnAngularReturnType {
   moduleMetadata?: NgModuleMetadata;
   template?: string;
   styles?: string[];
-  hasCustomTemplate?: boolean;
+  userDefinedTemplate?: boolean;
```

Comment:
can we do without adding this, and instead just look at the value of `template`?

Reply:
That's the problem, if template is missing, it's autogenerated. That's why I have to set this flag prior to the autogeneration...

Reply:
@shilman The source-decorator happens so late in the flow, and a template is required for rendering the component. I guess the template to be rendered could possibly be a more explicit property, leaving `template` as the actual template provided in the story. Don't really see how that would improve this particular code though.

Reply:
Gotcha, thanks for clarifying!